### ISSUE TITLE
Pdf Export doesnt work with  RS 8.6.12117

### ISSUE THREAD

**Issue by level-5** [ssuess/resourcespace-plugin-pdf_export#2]
Hi, 
i got the following message when try to generate a PDF /home/xxxx/public_html/xxxx/include/authenticate.php line 393: CSRF verification failed! Try reloading page.

In former Versions the Plugin works great.

Best Jan

**Comment by ssuess** [comment]
My initial guess would be that you have disabled http referrer sending (probably in the browser you are using, either by config or use of a plugin that does the same). Try re-enabling it or using another browser to see if that solves your problem.

**Comment by ssuess** [comment]
Another possibility is that RS has made code changes that do something similar, I will look into it further if the above does not solve your problem, let me know.

**Comment by level-5** [comment]
I found the following information on the RS site https://www.resourcespace.com/knowledge-base/systemadmin/csrf .

**Comment by ssuess** [comment]
Thanks for that link. I would first try setting `CSRF_enabled` to `false` and see if that fixes the problem. if it does, let me know (I don't have an 8.6 system setup for testing) and I will investigate further.

**Comment by level-5** [comment]
i set CSRF to false now i get the following message /home/xxxx/public_html/xxxx/lib/tcpdf/tcpdf.php line 8046: A non-numeric value encountered

**Comment by ssuess** [comment]
Ok, I will setup a test system and have a look at this. I will post back here once I have a fix/response.

**Comment by level-5** [comment]
Thanks

**Comment by ssuess** [comment]
Are you using the latest version (2.0) of the plugin? When I use that, and set `$CSRF_enabled = false;`, my system (8.6.12117 standard latest download from the RS website) will download a proper pdf when clicking on the create button.

**Comment by level-5** [comment]
Yes, i use the latest version. I found out that the second problem has to do with the PHP Version. With PHP 7.0 the pdf export works fine with 7.2 i got the message "lib/tcpdf/tcpdf.php line 8046: A non-numeric value encountered" see also here for that problem: https://www.drupal.org/project/webform/issues/2857854 . Thanks and best regards jan

**Comment by ssuess** [comment]
Hi there, I put in a fix for this that you can download from the master version, it allows CSRF_enabled to be set to true. However, it does not look as though previews are working for me (but create now works fine), can you tell me if you are correctly seeing previews in your setup?

**Comment by level-5** [comment]
Hi, the pdf export still only works for me when $CSRF_enabled = false and with php 7.0 - 
when $CSRF_enabled = true only the previewworks but not download.

with php 7.1 & 7.2 pdf export still dosent work - message (bomb)"lib/tcpdf/tcpdf.php line 8046: A non-numeric value encountered"

another problem is the export of a collection there i get the message (bomb): Sorry, an error has occurred. /plugins/pdf_export/pages/pdf_export_config.php line 205: Undefined offset: 0

**Comment by ssuess** [comment]
Hi, I have identified numerous issues that I am working through, apparently there have been significant changes and updates to various components and code since I released the last version 2 years ago. I will let you know when there is a new version available, but feel free to check the master branch as I am updating/fixing items as I find them in various files.

**Comment by ssuess** [comment]
One thing about your tcpdf error, this is not in my code, this is in the library. You might want to try a newer version of the tcpdf lib than the one installed by default with RS.

**Comment by ssuess** [comment]
and in the latest version that I just uploaded, both preview and create/download are working for me (php 7.0) and I have also fixed collections (the id was not sending because of a change that RS made). Check it out and let me know if it works for you.

**Comment by level-5** [comment]
thanks stephen, i'll test it out and let you know. it's a great plugin and
helps me a lot! best regards jan

Am So., 6. Jan. 2019 um 14:57 Uhr schrieb Stephen <notifications@github.com

**Comment by level-5** [comment]
The collection export now works for me. I try to export the Resource ID with the pdf but couldn t find a way also i got Problems saving a configuration - when try to generate a pdf from a saved preset it didn't work.                                                                                                                                                                                                                                                                                                                                         Gesendet von meinem BlackBerry 10-Smartphone.                                                                                                                                                                                                                Von: level.fiveGesendet: Dienstag, 8. Januar 2019 13:23An: ssuess/resourcespace-plugin-pdf_exportBetreff: Re: [ssuess/resourcespace-plugin-pdf_export] Pdf Export doesnt work with  RS 8.6.12117 (#2)thanks stephen, i'll test it out and let you know. it's a great plugin and helps me a lot! best regards janAm So., 6. Jan. 2019 um 14:57 Uhr schrieb Stephen <notifications@github.com>:and in the latest version that I just uploaded, both preview and create/download are working for me (php 7.0) and I have also fixed collections (the id was not sending because of a change that RS made). Check it out and let me know if it works for you.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by ssuess** [comment]
Thanks, I also have found that saving a config does not work while CSRF is enabled (it WILL work if disabled, so a workaround for now is to disable CSRF, save your configs, and then re-enable) I am working on a fix, but so far have not found a way to make CSRF work with an AJAX submit (which the save config command is).
